src/index.html

```diff
@@ -11,7 +11,202 @@
+          <input
+            type="number"
+            name="age"
+            id="age"
+            min="1"
+            max="100"
+            value="12"
+          >
+        </div>
+
+        <div class="form-field">
+          <label for="dateOfBirth">
+            Full date of birth
+          </label>
+          <input
+            type="date"
+            name="dateOfBirth"
+            id="dateOfBirth"
+          >
+        </div>
+
+        <div class="form-field">
+          <label for="agreement">
+            I accept the term of agreement
+          </label>
+          <input
+            type="checkbox"
+            name="agreement"
+            id="agreement"
+          >
+        </div>
+
+      </fieldset>
+
+      <fieldset class="form-fieldset">
+        <legend>Registration</legend>
+
+        <div class="form-field">
+          <label for="email">
+            E-mail:
+          </label>
+          <input
+            type="email"
+            name="email"
+            id="email"
+            placeholder="email@example.com"
+            required
+          >
+        </div>
+
+        <div class="form-field">
+          <label for="password">
+            Password:
+          </label>
+          <input
+            type="password"
+            name="password"
+            id="password"
+            required
+          >
+        </div>
+
+      </fieldset>
+
+      <fieldset class="form-fieldset">
+        <legend>An interesting fact about you!</legend>
+
+        <div class="form-field">
+          Do you love cats?
+          <label>
+            <input type="radio"
+            name="catLover"
+            value="yes"
+            >
+            Yes
+          </label>
+          <label>
+            <input type="radio"
+            name="catLover"
+            value="no"
+            >
+            No
+          </label>
+        </div>
+
+        <div class="form-field">
+          <label for="color">
+            What is your favorite color?
+          </label>
+          <input
+            type="color"
+            name="color"
+            id="color"
+          >
+        </div>
+
+        <div class="form-field">
+          <label for="sleepTime">
+            What time do you go to bed?
+          </label>
+          <input
+            type="time"
+            name="sleepTime"
+            id="sleepTime"
+          >
+        </div>
+
+        <div class="form-field">
+          <label for="favoriteCars">
+            What are your favorite brand of cars?
+          </label>
+          <select
+            name="favoriteCars"
+            id="favoriteCars"
+            multiple
+          >
+            <option value="bmw">BMW</option>
+            <option value="audi">Audi</option>
+            <option value="lada">Lada</option>
+          </select>
+        </div>
+
+        <div class="form-field">
+          <label for="rating">
+            How do you rate Our work?
+          </label>
+          <input
```

Comment:
It's good to set a minimal and maximal range for the input, if not for the seo then atleast for the code readability.